packages/grafana-ui/src/utils/processSeriesData.ts

```diff
@@ -212,3 +202,195 @@ export function sortSeriesData(data: SeriesData, sortIndex?: number, reverse = f
   }
   return data;
 }
+
+export class SeriesFieldProcessor {
+  series: SeriesData;
+  fields: Field[];
+  fieldIndexByName: { [key: string]: number };
+  timeFieldIndices: number[];
+  stringFieldIndices: number[];
+  numberFieldIndices: number[];
+  booleanFieldIndices: number[];
+  otherFieldIndices: number[];
```

Comment:
rather than a new array for each type, what about:
```
fieldTypeIndices: { [type: FieldType]: number[] };
```

and than adding FieldType as a parameter to the getters below?  That feels less brittle as we add/refine the various field type options and approaches.

Reply:
Good point